src/daemon/utils_cache.c

```diff
@@ -143,25 +148,29 @@ static int uc_insert(metric_t const *m, char const *key) {
   case DS_TYPE_COUNTER:
     ce->values_gauge = NAN;
     ce->values_raw.counter = m->value.counter;
-    ce->values_distribution = NULL;
+    ce->distribution_increase = NULL;
+    ce->start_value = create_typed_value(m->value, METRIC_TYPE_COUNTER);
```

Comment:
*create_typed_value()* doesn't seem to exist. Potentially this could be a composite literal like so:

```c
ce->start_value = (typed_value_t){m->value, METRIC_TYPE_COUNTER};
```

Reply:
Create a `typed_value_create` function. Using such composite literal here won't work as if type is distribution then we need to clone it